### PR TITLE
Set allowed versions of PHP to those still maintained

### PR DESCRIPTION
**Pull request by internalsystemerror**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | yes

### Description

This PR restricts the allowed versions of PHP to be `>=8.0` meaning only currently supported versions will be proposed in updates: https://www.php.net/supported-versions.php.

This will need to be updated yearly as support for older versions are dropped.

### CODE REVIEW

**Review by boesing — APPROVED** [review]
LGTM!

File: RENOVATE.md [hunk 1]
```diff
@@ -147,6 +147,7 @@ alerts, enabled by a previous preset, will use a `rangeStrategy` of `update-lock
     {"matchPackagePatterns": ["^mezzio/"], "groupSlug": "mezzio", "groupName": "all Mezzio packages"},
     {
         "matchPackageNames": ["php"],
+        "allowedVersions": ">=8.0"
```
Comment (Ocramius) [inline RENOVATE.md]: OK, but this isn't documented in our stuff :D 

Also, should this be `^8.0`? :thinking:

**Review by Ocramius — APPROVED** [review]

File: RENOVATE.md (hunk 1 above)
Comment (internalsystemerror) [reply RENOVATE.md]: The current description fits for this aswell, or I can break down exactly how this bit works instead if desired.

^8.0 works too, assuming this is updated each time, it shouldn't make a difference. I'll update though.

**Review by Ocramius — APPROVED** [review]
Let's try this out :)

**Pull request merged**